### PR TITLE
[BUGFIX] Regex angle brackets not HTML-escaped in Data Docs

### LINKED ISSUE #11907: Angle brackets in regular expressions are not escaped when rendering data docs
**Describe the bug**
When angle brackets are used in a regular expression for a ExpectColumnValuesToMatchRegex expectation, especially when combined with an exclamation mark for a negative lookbehind, these are passed directly into html, and ignored by the browser.

**To Reproduce**
Create an ExpectColumnValuesToMatchRegex expectation with the regular expression `^(?!\s)([A-Za-z0-9&()_\- \/*]+)(?<!\s)$`.
Run validation.
Generate and view data docs.

**Expected behavior**
Data docs include the `&lt;` html encoding, so browser correctly displays < when rendering the expectation.

**Environment (please complete the following information):**
 - Operating System: Any
 - Great Expectations Version: 1.17.2
 - Data Source: Any
 - Cloud environment: Databricks

**Additional context**
Please see screenshot below.  Note that the displated regex is truncated, and that the browser tools view shows the `<!` combination has been interpreted as an html comment.

Image

### PR DESCRIPTION
**Pull request by joshua-stauffer**
Closes #11907

## Summary

When a regex pattern containing angle-bracket characters (e.g. the negative lookbehind `(?<!\s)`) was used as an Expectation parameter, the angle brackets were emitted raw into the Data Docs HTML. The browser interpreted `<!` as the start of an HTML comment, silently truncating the rendered regex. This fix HTML-escapes all string parameter values before they are substituted into render templates so user-supplied content is displayed literally.

## Changes

- `great_expectations/render/view/view.py`: adds a `_escape_template_param_value` static method (using `html.escape`) and calls it on every param value in both code paths inside `render_string_template` — the `"styling"` branch (in-place dict mutation) and the plain-template branch (`escaped_params` dict) — before any `string.Template.safe_substitute` call. Surrounding markup (styling spans, base template string) is intentionally left raw.
- `tests/.../test_data_docs_regex_html_escaping.py`: new regression test that exercises angle brackets (`<`, `>`), ampersands (`&`), and the exact negative-lookbehind pattern from the bug report, covering both the styled and unstyled template paths.

## Why

`render_string_template` constructs Data Docs output by embedding param values directly into an HTML template string. Params are user-supplied (Expectation configuration) and may contain arbitrary characters. Characters like `<`, `>`, and `&` have special meaning in HTML; without escaping they silently corrupt the rendered output. HTML-escaping only the param values (not the surrounding markup) fixes the corruption without breaking existing styled output.

## User impact

Regexes and other Expectation parameter values containing `<`, `>`, or `&` now render correctly in Data Docs instead of being truncated or hidden. No change to the public API or serialized Expectation format; only the HTML rendering layer is affected.

## How to review

- The new test file (`test_data_docs_regex_html_escaping.py`) covers the reported pattern (`(?<!\s)`) plus edge cases (`>`, `&`, mixed). Run it with `pytest tests/render/test_data_docs_regex_html_escaping.py -v`.
- The two code paths in `render_string_template` (with and without `"styling"`) are exercised separately in the tests; check that both paths escape correctly.
- Existing render tests should be unaffected — escaped output for purely alphanumeric params is identical to the original.

### CODE REVIEW

File: great_expectations/render/view/view.py [hunk 1]
```diff
@@ -358,6 +374,13 @@ def render_string_template(self, template):  # noqa: C901, PLR0912 # FIXME CoP
         if "styling" in template:
             params = template.get("params", {})
 
+            # HTML-escape param values before any substitution so user-supplied
+            # content (e.g. a regex containing angle brackets) is rendered literally
+            # rather than interpreted as markup by the browser. Styling spans added
+            # below wrap the already-escaped content and remain raw HTML.
+            for parameter in params:
+                params[parameter] = self._escape_template_param_value(params[parameter])
```
[the remark this answers is not in the document]
Comment (joshua-stauffer) [reply great_expectations/render/view/view.py:397]: Fixed — `html_success_icon` is now excluded from escaping via a `_TRUSTED_HTML_PARAMS` set, so the status icon markup stays raw while user params are still escaped.

File: great_expectations/render/view/view.py [hunk 2]
```diff
@@ -404,14 +427,20 @@ def render_string_template(self, template):  # noqa: C901, PLR0912 # FIXME CoP
             ).safe_substitute(params)
             return string
 
+        # HTML-escape param values so user-supplied content (e.g. a regex containing
+        # angle brackets) is rendered literally rather than interpreted as markup.
+        escaped_params = {
+            parameter: self._escape_template_param_value(value)
+            for parameter, value in template.get("params", {}).items()
+        }
```
[the remark this answers is not in the document]
Comment (joshua-stauffer) [reply great_expectations/render/view/view.py:455]: Fixed — the unstyled branch now applies the same `_TRUSTED_HTML_PARAMS` exclusion, so `html_success_icon` is left raw here too.

File: tests/integration/data_sources_and_expectations/test_data_docs_regex_html_escaping.py [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+    validation_result = ExpectationSuiteValidationResult(
+        results=[
+            ExpectationValidationResult(
+                success=True,
+                expectation_config=ExpectationConfiguration(
+                    type="expect_column_values_to_match_regex",
+                    kwargs={
+                        "column": "my_col",
+                        "regex": REGEX_WITH_ANGLE_BRACKETS,
+                    },
```
[the remark this answers is not in the document]
Comment (joshua-stauffer) [reply tests/integration/data_sources_and_expectations/test_data_docs_regex_html_escaping.py]: Added `<`/`>`/`&` coverage, but through the **regex** param rather than the column name. The column name is rendered into the section header (`<h5>`) and TOC anchor via a separate path — not as a template param — so it isn't affected by this escaping change, and asserting on an escaped column name would fail. The regex param exercises all three characters through the param-escaping path the fix actually touches.

File: tests/integration/data_sources_and_expectations/test_data_docs_regex_html_escaping.py [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+    # if it does, "<!" is interpreted as the start of an HTML comment.
+    assert "(?<!" not in html, (
+        "Data Docs HTML contains an unescaped '<' from the regex negative "
+        "lookbehind; angle brackets in regular expressions should be "
+        "HTML-escaped so the browser renders them literally."
+    )
+    assert "(?&lt;!" in html, (
+        "Expected the regex angle bracket to be HTML-escaped as '&lt;' in the "
+        "rendered Data Docs output."
+    )
```
[the remark this answers is not in the document]
Comment (joshua-stauffer) [reply tests/integration/data_sources_and_expectations/test_data_docs_regex_html_escaping.py:103]: Done — the test now asserts `>` and `&` are escaped, that the trusted success icon is still emitted as raw HTML (not `&lt;i ...`), and adds a direct `render_string_template` call covering the unstyled branch.
